packages/app-store/office365calendar/api/callback.ts

```diff
@@ -74,6 +76,50 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
 
+  // Set the isDefaultCalendar as selectedCalendar
+  // If a user has multiple calendars, keep on making calls until we find the default calendar
+  let defaultCalendar: OfficeCalendar | undefined = undefined;
+  let requestUrl = "https://graph.microsoft.com/v1.0/me/calendars?$select=id,isDefaultCalendar";
+  let finishedParsingCalendars = false;
+
+  while (!finishedParsingCalendars) {
```

Comment:
MS Graph only returns 10 calendars at a time. If for any reason the primary calendar is not a part of the first call then continue making calendar calls.